File: RecyclerView/app/src/main/java/work/example/chris/recyclerview/MainActivity.java
package work.example.chris.recyclerview;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.List;

import work.example.chris.recyclerview.common.RecordAdapter;
import work.example.chris.recyclerview.model.Record;
import work.example.chris.recyclerview.utils.JsonHelper;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Initialize recyclerView
        initRecyclerView();
    }

    protected void onStart() {
        super.onStart();

        List<Record> recordList = JsonHelper.getModelFromRaw(this).getRecord();

        setupRecyclerView(recordList);
    }

    private void setupRecyclerView(List<Record> recordList) {
        RecordAdapter adapter = new RecordAdapter(recordList);
        recyclerView.setAdapter(adapter);
    }

    private void initRecyclerView() {
        recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
    }
}

File: HorizonRecyclerView/app/src/main/java/com/example/horizonrecyclerview/MainActivity.kt
package com.example.horizonrecyclerview

import android.os.Bundle
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.horizonrecyclerview.adapter.MyAdapter
import com.example.horizonrecyclerview.model.Version
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val version = ArrayList<Version>()
        version.addAll(Version.getList())
        recyclerView.layoutManager = LinearLayoutManager(this, LinearLayout.HORIZONTAL, false)
        recyclerView.adapter = MyAdapter(version)
    }
}
